src/Package/Fetch.zig

```diff
@@ -488,10 +485,13 @@ fn runResource(
             .include_paths = if (f.manifest) |m| m.paths else .{},
         };
 
-        // TODO:
-        // If any error occurred for files that were ultimately excluded, those
-        // errors should be ignored, such as failure to create symlinks that
-        // weren't supposed to be included anyway.
+        // Ignore errors that were excluded by manifest, such as failure to
+        // create symlinks that weren't supposed to be included anyway.
+        try unpack_result.filterErrors(filter);
+        if (unpack_result.hasErrors()) {
+            try unpack_result.bundleErrors(eb, try f.srcLoc(f.location_tok));
+            return error.FetchFailed;
+        }
```

Comment:
```suggestion
    try unpack_result.validate(eb, f.location_tok, filter);
```

Request to combine all that logic into a `validate` function.